drivers/gles3/storage/texture_storage.cpp

```diff
@@ -1553,6 +1604,15 @@ void TextureStorage::render_target_clear_used(RID p_render_target) {
 	rt->used_in_frame = false;
 }
 
+void TextureStorage::render_target_set_msaa(RID p_render_target, RS::ViewportMSAA p_msaa) {
+	RenderTarget *rt = render_target_owner.get_or_null(p_render_target);
+	ERR_FAIL_COND(!rt);
+#ifndef JAVASCRIPT_ENABLED
+	rt->msaa = p_msaa;
```

Comment:
We should likely have an error or warning if the user sets the 2D MSAA differently than the 3D MSAA as they share a framebuffer. 

Once we add Post effects that require a separate pass, we could remove the requirement for the MSAA to be the same and then properly resolve and copy the 3D color texture to the 2D Framebuffer. For now implementing that will be overkill.